### ISSUE TITLE
[Bug] Permissions.RequestAsync<Permissions.ContactsRead>() raises System.ObjectDisposedException

### ISSUE THREAD

**Issue by nbsoftware** [xamarin/Essentials#1373]
Permissions.RequestAsync<Permissions.ContactsRead> was working just fine until iOS 13.6

I have upgraded XCode to 11.7, built my app against latest 13.7 SDK and I got the exception System.ObjectDisposedException below, either on the Simulator or a physical device upgraded to 13.7

Tested again on an iOS 13.6 Simulator, works just fine as expected. 

```
System.ObjectDisposedException: Cannot access a disposed object.
  at AddressBook.ABAddressBook.AssertValid () [0x00012] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.22.1.8/src/Xamarin.iOS/AddressBook/ABAddressBook.cs:171 
  at AddressBook.ABAddressBook.get_Handle () [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.22.1.8/src/Xamarin.iOS/AddressBook/ABAddressBook.cs:176 
  at AddressBook.ABAddressBook.RequestAccess (System.Action`2[T1,T2] onCompleted) [0x00027] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.22.1.8/src/Xamarin.iOS/AddressBook/ABAddressBook.cs:204 
  at Xamarin.Essentials.Permissions+ContactsRead.RequestAddressBookPermission () [0x00016] in d:\a\1\s\Xamarin.Essentials\Permissions\Permissions.ios.cs:112 
  at Xamarin.Essentials.Permissions+ContactsRead.RequestAsync () [0x0001d] in d:\a\1\s\Xamarin.Essentials\Permissions\Permissions.ios.cs:91 
  at Xamarin.Essentials.Permissions.RequestAsync[TPermission] () [0x00000] in d:\a\1\s\Xamarin.Essentials\Permissions\Permissions.shared.cs:13 
  at fipacapp.Misc.PermissionsHelpers+<>c.<CheckAddressBookPermissions>b__1_0 () [0x0000f] 
...
```

Not too sure if this is a Xamarin.Essentials bug or a Xamarin.iOS one... so I'll post it to Xamarin.iOS too

### Basic Information

```
Visual Studio Community 2019 for Mac (Preview)
Version 8.8 Preview (8.8 build 1532)
Installation UUID: be4088b1-069a-44b1-bca1-a8bc61ab68dc
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000090

Mono Framework MDK
Runtime:
	Mono 6.12.0.90 (2020-02/d3daacdaa80) (64-bit)
	Package version: 612000090

...

Apple Developer Tools
Xcode 11.7 (16142)
Build 11E801a

Xamarin.Mac
Version: 6.22.1.8 (Visual Studio Community)
Hash: 6f5adef8e
Branch: d16-8
Build date: 2020-07-29 02:26:57-0400

Xamarin.iOS
Version: 13.22.1.8 (Visual Studio Community)
Hash: 6f5adef8e
Branch: d16-8
Build date: 2020-07-29 02:26:58-0400

Xamarin.Android
Version: 11.0.99.34 (Visual Studio Community)
Commit: xamarin-android/d16-8/49ba90e
Android SDK: /Users/fmeyer/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		8.1 (API level 27)
```

**Comment by nbsoftware** [comment]
Hello,

Well it seems like it is indeed a Xamarin Essentials bug after all.

Please see https://github.com/xamarin/xamarin-macios/issues/9580#issuecomment-688418884

**Comment by jamesmontemagno** [comment]
GC is probably being aggressive :( https://github.com/xamarin/Essentials/pull/1378/files may fix it up

**Comment by Redth** [comment]
Should be fixed in #1380
